### ISSUE TITLE
BAD BUG REMAINED OPEN!

### ISSUE THREAD

**Issue by xtronoid** [Microsoft/vscode#58010]
Issue Type: <b>Bug</b>

THIS ISSUE WAS BAD INTERPRETED BY EVERYONE:
https://github.com/Microsoft/vscode/issues/39988
connected with this
https://github.com/Microsoft/vscode/issues/15949

what i want to say is that isn´t that when opening a xml file for example via explorer, its wrong that an empty workspace is opened... BUT, the question is that EVEN if u close it, and reopen via ICON, ppl want that the workspace that it always used to be get back including open windows tabs preferrably.

in the first topic that settings is related with multiple IDE WINDOWS, not document tabs

and the second tab hotfix setting seems nothing to do with this for me too...

i think that one possible mid term solution (that dont breaks anything) could it be... when opening a single file in explorer... uses a NEW window, with empty workspace...

plz fix that! :D

Thanks.

VS Code version: Code 1.27.0 (7b9afc4196bda60b0facdf62cfc02815509b23d6, 2018-09-05T05:29:44.670Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-3610QM CPU @ 2.30GHz (8 x 2295)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|7.88GB (2.72GB free)|
|Process Argv|C:\Users\Xtronoid\AppData\Local\Programs\Microsoft VS Code\Code.exe|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (9)</summary>

Extension|Author (truncated)|Version
---|---|---
overtype|ada|0.2.0
npm-intellisense|chr|1.3.0
xml|Dot|2.3.2
beautify|Hoo|1.4.2
Angular2|joh|6.1.5
team|ms-|1.136.0
debugger-for-chrome|msj|4.9.1
nativescript|Tel|0.8.0
nativescript-ng2-snippets|www|2.0.1

</details>

**Comment by mjbvz** [comment]
Sorry but this is not readable. Please follow [this guide](https://github.com/Microsoft/vscode/wiki/Submitting-Bugs-and-Suggestions#writing-good-bug-reports-and-feature-requests) to file a proper issue report. Start with:

1. Steps to reproduce the proble,
1. What you expected to happen.
1. What actually happened.

**Comment by xtronoid** [comment]
1. -> open vs code via associated file in explorer
2. -> i want new window with empty workspace (for not breaking some compatibilites as mentioned in that posts, dont need to be the same one)
3. -> watch it:
           a) replace your workspace with an empty one
           b) or - if not already open a window with your workspace, it will open a window with empty workspace and even after you close that, and reopen via icon u are not in your workspace anymore

Pretty Anyonning,
Thanks.

**Comment by bpasero** [comment]
Sorry I do not understand. You can configure `window.openFilesInNewWindow: on` to force open files in separate windows. Does that help?

**Comment by xtronoid** [comment]
Hi there!

Well, this helps surely! thanks! But... if you dont already have an open window, it will change it to the empty workspace, even opening by icon later...

**Comment by bpasero** [comment]
@xtronoid so you want to restore the last opened workspace when opening a single file via doubleclick and add it there?

**Comment by xtronoid** [comment]
no...
i meant that if there´s no one open window before... it makes the empty the default one forever... got it?

**Comment by bpasero** [comment]
@xtronoid what do you mean by "default"? You mean it restores itself next time you just start VSCode?

**Comment by xtronoid** [comment]
let me help u solve it:
- An empty (new) workspace (that was created and switched to via click on file) dont have to got precedence over your saved and nice workspace! simple 💯

**Comment by xtronoid** [comment]
about your question... yes it restores the new empty each time you open via icon after, one time opened by file

**Comment by bpasero** [comment]
@xtronoid so to recap, you suggest to:
* if no window is open and you open a file, open it within the last workspace that was opened
* if a window is open and you open a file, open it in a new empty window

**Comment by xtronoid** [comment]
the second item u said now, was fixed creating in another one empty (the settings you helped me)

but the first item.... no... the first item... is a suggestion that someone said that can broken somethings like git or i dunno what...

so i suggested a fix to the first item that´s less intrusive... that´s as i said:

im saying about not what i want. but what´s happening:

if you got no one window open with your workspace,
then click on file,
it opens in a new empty workspace OK <- would be cooler yet if it mantains your actual workspace, BUT if it brokens something, it will be a bad choice.

so again... i suggest that AT LEAST:

afterrrrr you opent the file, after closssssed, then when you open it again via icon... it dont goes to the empty as a default forever! (but back to YOUR WORKSPACE! and not the new not saved one) got it?? lol

**Comment by bpasero** [comment]
@xtronoid how does that make any sense though, it seems to me that introduces very unpredictable behaviour for the user.

**Comment by bpasero** [comment]
None taken.

**Comment by xtronoid** [comment]
u can close it, but i just sayin that the empty one when u click via file, must not turn to be the default one! so u can click on the icon shortcut and starts normally back in your workspace...
or if isn´t really a problem, open in the same workspace and not in the new....

so simple

**Comment by darkmantle** [comment]
If it helps, what I think he means is this.

1: You close a workspace. Normally when you re-open Code it opens the last used workspace.

2: You then open a single file Code by using the file explorer context menu, but this should open in an empty workspace.

3: When you close the 2nd workspace above and open VS Code normally through the start menu / icon, it should open the 1st workspace again.

Basically opening a file via context menu should not override the "last opened workspace".
